external/ExternalIntegrationExample/ExternalIntegrationExample.cpp

```diff
@@ -0,0 +1,123 @@
+//*********************************************************************
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root
+// for license information.
+//
+// @File: ExternalIntegrationExample.cpp
+//
+// Purpose:
+//  This is a simple external C++ project main function file.
+//
+// Notes:
+//      <special-instructions>
+//
+//*********************************************************************
+
+#include "include/Common.h"
+
+// Include platform specific implementations of some MLOS functions.
+// Only needed in one compilation unit for a given target.
+//
+#include "MlosPlatform.Std.inl"
+
+#define IGNORE_UNUSED_VAR(x) (void)x
+
+void ThrowIfFail(HRESULT hr)
+{
+    if (FAILED(hr))
+    {
+        throw std::exception();
+    }
+}
+
+int main()
+{
+    // Create the MlosContext.
+    //
+    Mlos::Core::InterProcessMlosContextInitializer mlosContextInitializer;
+    HRESULT hr = mlosContextInitializer.Initialize();
+    ThrowIfFail(hr);
+
+    Mlos::Core::InterProcessMlosContext mlosContext(std::move(mlosContextInitializer));
+
+    // Create a feedback channel receiver thread.
+    //
+    ISharedChannel& feedbackChannel = mlosContext.FeedbackChannel();
+
+    // This background thread uses a lambda to monitor the feedback channel for
+    // new messages and process them using the callbacks registered for each
+    // message type in the global dispatch table.
+    //
+    std::future<bool> feedbackChannelReader = std::async(
```

Comment:
we should switch to std::thread

Reply:
This file is basically just a stripped down copy of `SmartCache/Main.cpp`.  I could change this one or we could leave that for a future PR.
I wasn't really aiming to test the C++ code itself - I feel like we have better more well documented examples of that already.  This was just to get us started with making sure we could reference all of the MLOS code enought to build/test it with a non-MLOS project.